cmd/spire-server/cli/run/run.go

```diff
@@ -397,3 +426,30 @@ func defaultConfig() *config {
 		},
 	}
 }
+
+type maybeBoolValue struct {
+	p **bool
+}
+
+func newMaybeBoolValue(p **bool) maybeBoolValue {
+	return maybeBoolValue{p: p}
+}
+
+func (b maybeBoolValue) Set(s string) error {
+	v, err := strconv.ParseBool(s)
+	if err != nil {
+		err = errors.New("parse error")
+	}
+	*b.p = &v
+	return err
+}
+
+func (b maybeBoolValue) String() string {
+	var v bool
+	if *b.p != nil {
```

Comment:
I believe we should also check if `b.p` is not nil. That is the case for`spire-server run -h`
```
	if b.p != nil && *b.p != nil {
```

Reply:
Hmm, that's surprising to me since `b.p` is unconditionally initialized with a pointer to the `c.UpstreamBundle` bool. Nevertheless I have also reproduced this. I'm gonna dig a bit so I can understand what is happening (it's like the Var is cloned or something).

Reply:
Ah, ok. It constructs a zero-value of the flag type and compares the String() method to determine if the flag is unset. Will fix.